docs/reference/pages/model_reference.rst

```diff
@@ -484,25 +484,43 @@ The ``locale`` and ``translation_key`` fields have a unique key constraint to pr
     .. autoattribute:: localized
 
 
-.. _page-revision-model-ref:
+.. _revision-model-ref:
 
-``PageRevision``
+``Revision``
 ================
 
-Every time a page is edited a new ``PageRevision`` is created and saved to the database. It can be used to find the full history of all changes that have been made to a page and it also provides a place for new changes to be kept before going live.
+Every time a page is edited a new ``Revision`` is created and saved to the database. It can be used to find the full history of all changes that have been made to a page and it also provides a place for new changes to be kept before going live.
 
 - Revisions can be created from any :class:`~wagtail.models.Page` object by calling its :meth:`~Page.save_revision` method
-- The content of the page is JSON-serialisable and stored in the :attr:`~PageRevision.content` field
-- You can retrieve a ``PageRevision`` as a :class:`~wagtail.models.Page` object by calling the :meth:`~PageRevision.as_page_object` method
+- The content of the page is JSON-serialisable and stored in the :attr:`~Revision.content` field
+- You can retrieve a ``Revision`` as a :class:`~wagtail.models.Page` object by calling the :meth:`~Revision.as_page_object` method
+
+.. versionchanged:: 4.0
+
+    The model has been renamed from ``PageRevision`` to ``Revision`` and it now references the ``Page`` model using a combination of an ``object_id`` :class:`~django.db.models.CharField` and foreign keys to :class:`~django.contrib.contenttypes.models.ContentType`.
 
 Database fields
 ~~~~~~~~~~~~~~~
 
-.. class:: PageRevision
+.. class:: Revision
 
-    .. attribute:: page
+    .. attribute:: content_type
 
-        (foreign key to :class:`~wagtail.models.Page`)
+        (foreign key to :class:`~django.contrib.contenttypes.models.ContentType`)
+
+        This is the content type of the object this revision belongs to. For page revisions, this means the content type of the specific page type.
+
+    .. attribute:: base_content_type
+
+        (foreign key to :class:`~django.contrib.contenttypes.models.ContentType`)
+
+        This is the base content type of the object this revision belongs to. For page revisions, this means the content type of the default page type.
+
+    .. attribute:: object_id
+
+        (string)
+
+        This represents the id of the object this revision belongs to.
```

Comment:
I think we should use the term "primary key" since this may not always be a traditional "id" field:

> This represents the primary key of the object this revision belongs to.